app/models/user.js

```diff
@@ -0,0 +1,58 @@
+const bcrypt = require('bcrypt');
+
+const User = (sequelize, DataTypes) => {
+  sequelize.define(
+    'user',
+    {
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        validate: {
+          notEmpty: true,
+          len: [1, 50]
+        }
+      },
+      lastName: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        validate: {
+          notEmpty: true,
+          len: [1, 50]
+        }
+      },
+      email: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        validate: {
+          isEmail: true,
+          notEmpty: true,
+          len: [1, 255]
+        }
+      },
+      password: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        validate: {
+          notEmpty: true
+        }
+      }
+    },
+    {
+      freezeTableName: true,
+      indexes: [{ unique: true, fields: ['email'] }],
+      hooks: {
+        beforeCreate: user => {
+          const salt = bcrypt.genSaltSync();
+          user.password = bcrypt.hashSync(user.password, salt);
+        }
+      },
+      instanceMethods: {
+        validPassword(password) {
+          return bcrypt.compareSync(password, this.password);
+        }
+      }
```

Comment:
This is ok but, we normally don't put logic inside of models. Also some may argue that using the synchronous version of those functions isn't good because of the blocking. You can read more [here](https://nodejs.org/de/docs/guides/blocking-vs-non-blocking/)

So, try moving these functions maybe to a helper module.